consensus/spos/bls/subroundSignature.go

```diff
@@ -368,7 +363,19 @@ func (sr *subroundSignature) doSignatureJobForManagedKeys() bool {
 		}
 
 		managedPrivateKey := sr.GetMessageSigningPrivateKey(pkBytes)
-		signatureShare, err := sr.MultiSigner().CreateAndAddSignatureShareForKey(sr.GetData(), managedPrivateKey, pkBytes)
+		selfIndex, err := sr.ConsensusGroupIndex(pk)
+		if err != nil {
+			log.Warn("doSignatureJobForManagedKeys: index not found", "pk", pkBytes)
+			continue
+		}
+
+		managedPrivateKeyBytes, err := managedPrivateKey.ToByteArray()
```

Comment:
signature handler was supposed to do itself the management of private keys and signing and move that out of the consensus, but here it is partially moved into consensus again.
Maybe add a todo and I will do the refactor, or you can move the management into the signature handler.

Reply:
added TODO